tests/test_viewer.py

```diff
@@ -319,19 +320,23 @@ def test_browser(self):
         finally:
             os.remove(f.name)
 
+    def basic_judgment(self):
```

Comment:
This is indeed a repeating pattern. However, this is more like a common sequential test of a larger sequential test. You can tell that the tests after this common piece is similar but not the same. So abstrating this piece out won't make the test easier to understand(for example, it's not obvious why the test should listen to port 9003).

It is not uncommon for **testcases** to have a common pattern, we don't need to abstract them all.

Reply:
I get it